src/jshint.js

```diff
@@ -401,8 +401,35 @@ var JSHINT = (function() {
     return i;
   }
 
-  function doOption() {
-    var nt = state.tokens.next;
+  function doLexOption(nt) {
```

Comment:
I don't like having to add a doLexOption but I don't see a choice to make `maxlen` backwards compatible because lexer options are applied a token before. There are possibly ways to do this with a bigger refactor, but since maxlen will be removed in the next major, my plan was do this for now and remove this function along with maxlen.